package/kedro_viz/models/flowchart.py

```diff
@@ -769,35 +723,39 @@ def set_run_command(cls, _):
             return f"kedro run --to-outputs={cls.data_node.name}"
         return None
 
-    @validator("plot", always=True)
-    def set_plot(cls, _):
-        if cls.data_node.is_plot_node():
-            return cls.data_node.kedro_obj.load()
-        return None
+    @validator("preview", always=True)
+    def set_preview(cls, _):
+        if cls.data_node.is_preview_disabled() or not hasattr(cls.dataset, "preview"):
+            return None
 
-    @validator("image", always=True)
-    def set_image(cls, _):
-        if cls.data_node.is_image_node():
-            return cls.data_node.kedro_obj.load()
-        return None
+        try:
+            preview_args = (
+                cls.data_node.get_preview_args() if cls.data_node.viz_metadata else None
+            )
+            if preview_args is None:
+                return cls.dataset.preview()
```

Comment:
ok so if the user does the below 

```
companies:
  type: pandas.CSVDataset
  filepath: ${_base_location}/01_raw/companies.csv
```

if self.viz_metadata doesn't exist, the preview is still enabled also because there are no preview_args which is required only for TablePreview, it defaults it to 5. 

since there is no `viz_metadata` and hence no `preview_args` ... get_preview_args() will throw an error because it tries to access self.viz_metadata which doesn't exist. hence only if the user has specified viz_metadata, we allow `preview_args` to load.

Reply:
ok so if the user does the below

companies:
  type: pandas.CSVDataset
  filepath: ${_base_location}/01_raw/companies.csv


Preview remains enabled by default, displaying only the first 5 rows. Preview is disabled only if the user explicitly sets preview: false

In the provided configuration, self.viz_metadata does not exist, the `is_preview_disabled()` returns false which means preview is enable. However, because there are no preview_args, it defaults to displaying 5 rows.


```
 preview_args = (
                cls.data_node.get_preview_args() if cls.data_node.viz_metadata else None
            )
```

In the above code, since there is no viz_metadata, and therefore no preview_args, calling get_preview_args() will throw an error because it attempts to access self.viz_metadata, which does not exist. Therefore, we only allow preview_args to load if the user has specified viz_metadata.